File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temp) {
        int ak[] = new int [temp.length];
        Stack<Integer> m = new Stack<>();
        for(int i = 0; i < temp.length; i++){
            while(!m.isEmpty() && temp[i] > temp[m.peek()]){
                int index = m.pop();
                ak[index] = i - index;
            }
            m.push(i);
        }
         return ak;
    }
}

File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        vector<int> ak(temp.size());
        stack<int> mj;
        for(int i = 0; i<temp.size(); i++){
            while(!mj.empty() && temp[i] > temp[mj.top()]){
                int index = mj.top();
                ak[index] = i - index;
                mj.pop();
            }
            mj.push(i);
        }
        return ak; 
    }
};